ci/build_conda.sh

```diff
@@ -15,6 +15,16 @@ rapids-print-env
 
 rapids-logger "Begin py build"
 
-rapids-conda-retry mambabuild conda/recipes/pynvjitlink
+export CUDA_VERSION="$(cat pynvjitlink/CUDA_VERSION)"
+
+cat > cuda_compiler_version.yaml << EOF
+cuda_compiler_version:
+  - "${CUDA_VERSION}"
+EOF
+
+rapids-conda-retry build \
+    conda/recipes/pynvjitlink \
+    -m cuda_compiler_version.yaml \
+;
```

Comment:
Then by passing the variant to build, it is included in the hash output of the package as shown at the end of the build [on CI]( https://github.com/rapidsai/pynvjitlink/actions/runs/10259035969/job/28382860270?pr=97#step:7:735 ):

```json
INFO :: The inputs making up the hashes for the built packages are as follows:
{
  "pynvjitlink-0.3.0-py311hbec61cd_0.tar.bz2": {
    "recipe": {
      "c_compiler": "gcc",
      "c_compiler_version": "11",
      "c_stdlib": "sysroot",
      "c_stdlib_version": "2.17",
      "cuda_compiler": "cuda-nvcc",
      "cuda_compiler_version": "12.6",
      "cxx_compiler": "gxx",
      "cxx_compiler_version": "11",
      "target_platform": "linux-aarch64"
    }
  }
}
```